demo-snippet.html

```diff
@@ -134,16 +126,18 @@
 
         // If there's no template, render empty code.
         if (!template) {
-          this._markdown = '```\n```';
+          this._markdown = '';
           return;
         }
 
-        var snippet = this.$.marked.unindent(template.innerHTML);
+        var snippet = template.innerHTML;
 
         // Boolean properties are displayed as checked="", so remove the ="" bit.
         snippet = snippet.replace(/=""/g, '');
+        this._markdown = this._unindent(snippet);
 
-        this._markdown = '```\n' + snippet + '\n' + '```';
+        // Remove any leading/trailing new lines.
+        this._markdown = this._markdown.replace(/^\n*|\n*$/g, '');
```

Comment:
Since there is a binding on this property, could you keep updating `snippet` and assign it to `this._markdown` only once?